app/src/main/java/com/bnyro/recorder/ui/components/PlayerController.kt

```diff
@@ -0,0 +1,185 @@
+                        }
+
+                        PlayerState.Play -> {
+                            Icon(
+                                Icons.Default.Pause,
+                                contentDescription = "Pause"
+                            )
+                        }
+
+                        PlayerState.Pause -> {
+                            Icon(
+                                Icons.Default.PlayArrow,
+                                contentDescription = "Play"
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Player.isPlayingState(): State<PlayerState> {
+    return produceState(
+        initialValue = if (isPlaying) {
+            PlayerState.Play
+        } else {
+            PlayerState.Pause
+        },
+        this
+    ) {
+        val listener = object : Player.Listener {
+            override fun onIsPlayingChanged(isPlaying: Boolean) {
+                playbackState
+                value = if (isPlaying) {
+                    PlayerState.Play
+                } else {
+                    PlayerState.Pause
+                }
+            }
+        }
+        addListener(listener)
+        if (!isActive) {
+            removeListener(listener)
+        }
+    }
+}
+
+@Composable
+fun Player.positionAndDurationState(): State<Pair<Long, Long?>> {
+    return produceState(
+        initialValue = (currentPosition to duration.let { if (it < 0) null else it }),
+        this
+    ) {
+        var isSeeking = false
+        val listener = object : Player.Listener {
+            override fun onPlaybackStateChanged(playbackState: Int) {
+                if (playbackState == Player.STATE_READY) {
+                    isSeeking = false
+                }
+            }
+
+            override fun onMediaItemTransition(mediaItem: MediaItem?, reason: Int) {
+                value = currentPosition to value.second
+            }
+
+            override fun onPositionDiscontinuity(
+                oldPosition: Player.PositionInfo,
+                newPosition: Player.PositionInfo,
+                reason: Int
+            ) {
+                if (reason == Player.DISCONTINUITY_REASON_SEEK) {
+                    isSeeking = true
+                    value = currentPosition to duration.let { if (it < 0) null else it }
+                }
+            }
+        }
+        addListener(listener)
+
+        val pollJob = launch {
+            while (isActive) {
+                delay(1000)
+                if (!isSeeking) {
+                    value = currentPosition to duration.let { if (it < 0) null else it }
+                }
+            }
+        }
+        if (!isActive) {
+            pollJob.cancel()
+            removeListener(listener)
+        }
+    }
+}
+
+fun Player.playPause() {
+    if (isPlaying) pause() else play()
+}
+
+enum class PlayerState {
+    Buffer, Play, Pause
```

Comment:
I'd prefer using `Buffering, Playing, Paused` as enum names, since that's the naming scheme `ExoPlayer` uses as well.